src/reflog.rs

```diff
@@ -0,0 +1,379 @@
+  }
+
+  #[napi]
+  /// Get the message bytes of this reflog entry.
+  ///
+  /// @category ReflogEntry/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class ReflogEntry {
+  ///   messageBytes(): Uint8Array | null;
+  /// }
+  /// ```
+  ///
+  /// @returns Message bytes of this reflog entry. Returns `null` if no message is present.
+  pub fn message_bytes(&self) -> Option<Uint8Array> {
+    self.inner.message_bytes().map(|bytes| bytes.to_vec().into())
+  }
+}
+
+#[napi(iterator)]
+/// An iterator over the entries in a reflog.
+pub struct ReflogIter {
+  pub(crate) inner: SharedReference<Reflog, git2::ReflogIter<'static>>,
+}
+
+#[napi]
+impl Generator for ReflogIter {
+  type Yield = ReflogEntry;
+  type Next = ();
+  type Return = ();
+
+  fn next(&mut self, _value: Option<Self::Next>) -> Option<Self::Yield> {
+    self.inner.next().map(|entry| ReflogEntry {
+      inner: ReflogEntryInner::Owned(entry),
+    })
+  }
+}
+
+/// Internal implementation of Reflog holding the original git2 value
+pub(crate) enum ReflogInner {
+  Owned(git2::Reflog),
+}
+
+impl Deref for ReflogInner {
+  type Target = git2::Reflog;
+
+  fn deref(&self) -> &Self::Target {
+    match self {
+      Self::Owned(inner) => inner,
+    }
+  }
+}
+
+impl DerefMut for ReflogInner {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    match self {
+      Self::Owned(inner) => inner,
+    }
+  }
+}
+
+#[napi]
+/// A class to represent a git reflog.
+pub struct Reflog {
+  pub(crate) inner: ReflogInner,
+}
+
+#[napi]
+impl Reflog {
+  #[napi]
+  /// Append a new entry to the reflog.
+  ///
+  /// @category Reflog/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Reflog {
+  ///   append(newOid: string, committer: Signature, msg?: string | null | undefined): void;
+  /// }
+  /// ```
+  ///
+  /// @param {string} newOid - New object ID (SHA1) for this reflog entry.
+  /// @param {Signature} committer - Committer signature for this reflog entry.
+  /// @param {string} [msg] - Optional message for this reflog entry.
+  /// @throws Throws error if the OID is invalid or if appending fails.
+  pub fn append(&mut self, new_oid: String, committer: Signature, msg: Option<String>) -> crate::Result<()> {
+    let git2_new_oid = Oid::from_str(&new_oid)?;
+    let git2_committer: git2::Signature<'_> = git2::Signature::try_from(committer)?;
+    self.inner.append(git2_new_oid, &git2_committer, msg.as_deref())?;
+    Ok(())
+  }
+
+  #[napi]
+  /// Remove an entry from the reflog.
+  ///
+  /// @category Reflog/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Reflog {
+  ///   remove(i: number, rewritePreviousEntry: boolean): void;
+  /// }
+  /// ```
+  ///
+  /// @param {number} i - Index of the entry to remove.
+  /// @param {boolean} rewritePreviousEntry - Whether to rewrite the previous entry.
+  /// @throws Throws error if the index is invalid or if removal fails.
+  pub fn remove(&mut self, i: u32, rewrite_previous_entry: bool) -> crate::Result<()> {
```

Comment:
```suggestion
  pub fn remove(&mut self, i: u32, rewrite_previous_entry: Option<bool>) -> crate::Result<()> {
```
Since libgit2 expect `rewrite_previous_entry` default value as 0, so how about receiving it as an optional value and handling the default value internally?

Reply:
I missed that. I’ll change it to “Option<bool>” and handle the default internally to match git2’s behavior. Thanks!